src/prisma/schedule.ts

```diff
@@ -37,3 +39,108 @@ async function initiateSchedule(scheduleyear: number) {
   });
   return schedule;
 }
+
+async function updateCurrentSchedule(
+  scheduleId: any,
+  courses: CourseSectionInput[]
+) {
+  // Simply creates a new schedule instead of updating the current schedule
+
+  try {
+    const currentSchedule = await prisma.schedule.findFirst({
+      where: {
+        id: parseInt(scheduleId),
+      },
+    });
+
+    if (currentSchedule) {
+      // Delete course sections linked to original schedule
+      await prisma.courseSection.deleteMany({
+        where: {
+          scheduleId: parseInt(scheduleId),
+        },
+      });
+
+      // Delete original schedule
+      await prisma.schedule.delete({
+        where: {
+          id: parseInt(scheduleId),
+        },
+      });
+    }
+  } catch (error) {
+    return {
+      success: false,
+      message: 'Error deleting original schedule',
+      errors: error,
+    };
+  }
+
+  const currentYear = new Date().getFullYear();
+  let newSchedule;
+
+  try {
+    newSchedule = await prisma.schedule.create({
+      data: {
+        year: currentYear,
+      },
+    });
```

Comment:
There shouldn't be a situation where we create new schedule entry as that would result in having to assume a default year.

Reply:
Anyhow, I'm a bit conflicted on the exact behaviour. We could also do it so we pull the year from the most recent schedule (ie. no ID is given) and create a new entry using that year.

Reply:
Okay, I'm just going to implement it for both circumstances, If for some reason an Id isn't given, I'm gonna use the latest schedule, otherwise use the given id.